internal/utils/utils.go

```diff
@@ -0,0 +1,363 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	ecv1alpha1 "go.etcd.io/etcd-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	SSSmallerThanDesired = "SSSmallerThanDesired"
+	SSBiggerThanDesired  = "SSBiggerThanDesired"
+	SSEqualToDesired     = "SSEqualToDesired"
```

Comment:
Suggest to remove them for now. Also usually a better way to define an enum for such case.